Fixed for stripe-checkout parameter changes

ivgtr opened the pull request:
seems to Fix #117 

Change it to accept the new parameters.
This is a new way to use lineItems and price instead of items and SKUs.
SKUs are not supported, but they still work, so I've left the parameters in place.
If you want to merge these changes, it is likely that the READEME.md and Demosite will need to be modified as well.

leetworx commented:
Hi there, is it possible to cut an updated 3.4.2 release with this fix so as to align with the current stripe API? I saw mention made somewhere of being able to git clone and use the plugin from source, but I'm new to nuxt and haven't seen any examples of how one might go about that. Thanks!

jofftiquez commented on src/Checkout.vue:
```diff
@@ -19,7 +19,11 @@ export default {
-    items: {
+    mode:{
+      type: String,
+      required: undefined
```
Don't use undefined. If this prop is required, use true, otherwise, omit the `required` key.

jofftiquez commented on src/Checkout.vue:
```diff
@@ -70,8 +74,9 @@ export default {
               cancelUrl: this.cancelUrl,
               clientReferenceId: this.clientReferenceId,
               customerEmail: this.customerEmail,
-              items: this.items,
```
The `items` option is still present in [the docs](https://stripe.com/docs/js/checkout/redirect_to_checkout#stripe_checkout_redirect_to_checkout-options-items). Please do not omit this.

jofftiquez commented:
Hello, @ivgtr I have a few change requests for this PR. Thanks for your help.

ivgtr commented:
Hi @jofftiquez 
Thanks for making this plugin. I'm using it.

The mode parameter was not used in the previous Specifications,
so I'll set that value so that it's not required for backwards compatibility. (I specified it incorrectly, sorry)
And, I removed a parameter by mistake, so I fixed it.

Make sure you're okay with this fix.

jofftiquez commented on src/Checkout.vue:
```diff
@@ -19,10 +19,18 @@ export default {
     },
+    mode:{
+      type: String,
+      required: false
```
Remove if not required, the absence of the field `required` also means that the prop is not required. Please remove.

ivgtr replied:
The old request for `SKU` doesn't use `mode`, so it's not required, but the new one won't work without it.
I'm not sure if a request for `SKU` will work properly when `mode` is present. 
By the way, I can't verify this because I can't create a new SKU product. Umm.

jofftiquez commented:
Hello @ivgtr thank you for using this. :) 

One last comment and everything's good.

ivgtr commented:
The old request for SKU doesn't use mode, so it's not required, but the new one won't work without it.
I'm not sure if a request for SKU products will work when mode is present. so I'll specify the default.
By the way, I can't verify this because I can't create a new SKU product. Umm.
Also, it was correct to split the redirectToCheckout() argument according to the presence or absence of mode, so I fixed it.

Sorry to bother you again, Make sure you're okay with this fix.
